ta4j-core/src/main/java/org/ta4j/core/criteria/EnterAndHoldCriterion.java

```diff
@@ -53,29 +56,35 @@
  */
 public class EnterAndHoldCriterion extends AbstractAnalysisCriterion {
 
+    /** The amount to be used to hold the entry position */
+    private final BigDecimal amount;
     private final TradeType tradeType;
     private final AnalysisCriterion criterion;
 
-    /** The {@link ReturnCriterion} (with base) from a buy-and-hold strategy. */
+    /**
+     * The {@link ReturnCriterion} (with base) from a buy-and-hold strategy with an
+     * {@link #amount} of {@code 1}.
+     */
     public static EnterAndHoldCriterion EnterAndHoldReturnCriterion() {
         return new EnterAndHoldCriterion(TradeType.BUY, new ReturnCriterion());
     }
 
     /**
-     * Constructor for buy-and-hold strategy.
+     * Constructor for buy-and-hold strategy with an {@link #amount} of {@code 1}.
      *
      * @param criterion the {@link AnalysisCriterion criterion} to calculate
      * @throws IllegalArgumentException if {@code criterion} is an instance of
      *                                  {@code EnterAndHoldCriterion} or
      *                                  {@code VersusEnterAndHoldCriterion}
      */
     public EnterAndHoldCriterion(AnalysisCriterion criterion) {
+        this.amount = BigDecimal.ONE;
         this.tradeType = TradeType.BUY;
         this.criterion = criterion;
```

Comment:
```suggestion
        this(TradeType.BUY, criterion);
```

Reply:
My bad, I missed this one earlier. Convenience constructors should funnel into the "main" constructor rather than setting properties directly. This both avoids duplication and protects against possible future divergence. In this case there's additional input validation on the `criterion` param that's in the other constructor but not this. This leads to unwanted inconsistent behavior between the constructors.

Reply:
Re-read the sentence before what you quoted.